doi/src/main/java/ca/nrc/cadc/doi/DoiAction.java

```diff
@@ -205,7 +284,271 @@ private void parsePath() {
+                        log.error(String.format("Unable to parse requester uid[%s] for doi: %s",
+                                requester.getValue(), childNode.getName()), e);
+                    }
+                }
+            }
+        }
+        return ownedNodes;
+    }
+
+    protected void getStatusList(List<Node> nodes) throws Exception {
+        List<DoiStatus> doiStatusList = new ArrayList<>();
+        for (Node node : nodes) {
+            log.debug("StatusList node: " + node.getName());
+            // Verify this is a container node before continuing
+            if (node instanceof ContainerNode) {
+                try {
+                    ContainerNode doiContainerNode = vospaceDoiClient.getContainerNode(node.getName());
+                    DoiStatus doiStatus = getDoiStatus(node.getName(), doiContainerNode, false);
+                    doiStatusList.add(doiStatus);
+                    log.debug("added doiStatus: " + doiStatus);
+                } catch (Exception ex) {
+                    // skip
+                    log.debug(String.format("skipping %s because %s", node.getName(), ex.getMessage()));
+                }
+            } else {
+                log.warn("Non-container node found in DOI base directory. Skipping... ");
+            }
+        }
+        log.debug("doiStatusList size: " + doiStatusList.size());
+
+        String docFormat = this.syncInput.getHeader("Accept");
+        log.debug("'Accept' value in header is " + docFormat);
+        if (docFormat != null && docFormat.contains("application/json")) {
+            // json document
+            syncOutput.setHeader("Content-Type", "application/json");
+            DoiStatusListJsonWriter writer = new DoiStatusListJsonWriter();
+            writer.write(doiStatusList, syncOutput.getOutputStream());
+        } else {
+            // xml document
+            syncOutput.setHeader("Content-Type", "text/xml");
+            DoiStatusListXmlWriter writer = new DoiStatusListXmlWriter();
+            writer.write(doiStatusList, syncOutput.getOutputStream());
+        }
+    }
+
+    protected DoiStatus getDoiStatus(String doiSuffixString, ContainerNode doiContainerNode, boolean authorize) throws Exception {
+        DoiStatus doiStatus;
+        if (!authorize || vospaceDoiClient.isCallerAllowed(doiContainerNode, getAdminSubject())) {
+            // get status
+            String status = doiContainerNode.getPropertyValue(DOI_VOS_STATUS_PROP);
+            if (StringUtil.hasText(status)
+                    && !status.equals(Status.ERROR_REGISTERING.getValue())
+                    && !status.equals(Status.ERROR_LOCKING_DATA.getValue())) {
+                // update status based on the result of the minting service
+                status = updateMintingStatus(doiContainerNode, status);
+            }
+
+            // get the data directory
+            // TODO why do this when the data directory path is known???
+            String dataDirectory = null;
```

Comment:
As the code comment says, no need to search through the doi child nodes looking for the data node, just build the dataDirectory path as below and that's it.